RecoBTag/CTagging/src/CharmTagger.cc

```diff
@@ -0,0 +1,106 @@
+#include "RecoBTag/CTagging/interface/CharmTagger.h"
+
+#include "DataFormats/BTauReco/interface/CandSoftLeptonTagInfo.h"
+#include "DataFormats/BTauReco/interface/CandIPTagInfo.h"
+#include "DataFormats/BTauReco/interface/SecondaryVertexTagInfo.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "CondFormats/DataRecord/interface/GBRWrapperRcd.h"
+
+#include <iostream>
+#include <vector>
+#include <algorithm>
+#include <map>
+#include <iostream>
+
+CharmTagger::CharmTagger(const edm::ParameterSet & configuration):
+	sl_computer_(configuration.getParameter<edm::ParameterSet>("slComputerCfg")),
+	vars_definition_(configuration.getParameter<vpset>("variables")),
+  mva_name_( configuration.getParameter<std::string >("mvaName") ),
+  use_condDB_(configuration.getParameter<bool>("useCondDB")),
+  gbrForest_label_(configuration.getParameter<std::string>("gbrForestLabel")),
+  weight_file_(configuration.getParameter<edm::FileInPath>("weightFile")),
+  use_GBRForest_(configuration.getParameter<bool>("useGBRForest")),
+  use_adaBoost_(configuration.getParameter<bool>("useAdaBoost"))
+{
+	uses(0, "pfImpactParameterTagInfos");
+	uses(1, "pfInclusiveSecondaryVertexFinderCvsLTagInfos");
+	uses(2, "softPFMuonsTagInfos");
+	uses(3, "softPFElectronsTagInfos");
+}
+
+void CharmTagger::initialize(const JetTagComputerRecord & record)
+{
+	mvaID_.reset(new TMVAEvaluator());
+
+	std::vector<std::string> variable_names;
+	variable_names.reserve(vars_definition_.size());
+
+	for(auto &var : vars_definition_) {
+		variable_names.push_back(
+			var.getParameter<std::string>("name")
+			);
+
+		MVAVar mva_var;
+		mva_var.name = var.getParameter<std::string>("name");
+		mva_var.id = reco::getTaggingVariableName(
+			var.getParameter<std::string>("taggingVarName")
+			);
+		mva_var.has_index = var.existsAs<int>("idx") ;
+		mva_var.index = mva_var.has_index ? var.getParameter<int>("idx") : 0;
+		mva_var.default_value = var.getParameter<double>("default");
+
+		variables_.push_back(mva_var);
```

Comment:
vars_definition_ is used only to define variables_.
I think vars_definition_ should be eliminated and variables_ vector is to be filled in the constructor.